src/Apps/Order/Routes/Shipping2/Hooks/useShippingContext.ts

```diff
@@ -0,0 +1,173 @@
+import { createContext, useContext, useMemo, useState } from "react"
+import { FormikProps } from "formik"
+import { ShippingProps, ShippingRouteStep } from "Apps/Order/Routes/Shipping2"
+import {
+  FulfillmentValues,
+  ShipValues,
+} from "Apps/Order/Routes/Shipping2/FulfillmentDetails"
+import {
+  addressWithFallbackValues,
+  getDefaultUserAddress,
+  FulfillmentType,
+} from "Apps/Order/Routes/Shipping2/Utils/shippingUtils"
+import { extractNodes } from "Utils/extractNodes"
+import {
+  ParsedOrderData,
+  useParseOrderData,
+} from "Apps/Order/Routes/Shipping2/Hooks/useParseOrderData"
+
+type FulfillmentHelpers = Pick<
+  FormikProps<FulfillmentValues>,
+  "handleSubmit" | "isValid" | "values"
+>
+export interface ShippingContextProps {
+  parsedOrderData: ParsedOrderData
+  initialValues: {
+    fulfillmentDetails: FulfillmentValues
+    shippingQuotes: {
+      selectedShippingQuoteId?: string | null
+    }
+  }
+  step: ShippingRouteStep
+  helpers: {
+    fulfillmentDetails: FulfillmentHelpers & {
+      setFulfillmentFormHelpers: (helpers: FulfillmentHelpers) => void
+    }
+  }
+}
+
+export const ShippingContext = createContext<ShippingContextProps>({} as any)
```

Comment:
I can understand keeping the context itself inside of the useShippingContext hook, but this is unconventional across the app. Contexts should declare themselves -- `Shipping2/ShippingContext.tsx`, and we can either add an additional hook inside of `Shipping2/Hooks/useShippingContext` (which returns the context), or just have the hook exported directly from `ShippingContext`. We do it both ways.

Reply:
Also, no `as any` here. Its better to cast as `unknown` first, and then as the correct type:

Reply:
Thanks and yeah, I wondered about this, ultimately decided to stay simple for now since ShippingContext.Provider is *technically* a component, but in practice is mostly about the hook. Will move out.